content/docs/base-components/ReactiveComponent.md

```diff
@@ -15,50 +15,110 @@ redirect_from:
+Let's suppose - we are building an e-commerce store for cars which displays a list of cars and 5 popular car brands in tabs UI as the user selects a tab the result gets updated with the cars of selected brand. Now if the user wants to select other brand from tab, the results should be updated to new selected brand. In this case, `ReactiveComponent` can be used with `defaultQuery` to achieve the desired behavior easily.
 
-We can then use the given ReactiveComponent to be watched by all the filters (via `react` prop) to avail the desired brand based filtering for all the filters.
+We can then use the given ReactiveComponent to be watched by the result component (via `react` prop) to avail the desired brand based filtering for the results.
 
-Check demo [here](https://codesandbox.io/s/3ylrrr0r5q).
+Check demo [here](https://codesandbox.io/s/7zrj740oj6).
 
 
 ```js
  <ReactiveComponent
-    componentId="CarSensor"
-    customQuery={props => ({
-        query: {
-                term: {
-                    "brand.keyword": "Ford"
-                }
-            }
+    componentId="Brand"
+    showFilter
+    defaultQuery={() => ({
+        aggs: {
+            'brand.keyword': {
+                terms: {
+                    field: 'brand.keyword',
+                    order: {
+                        _count: 'desc',
+                    },
+                    size: 5,
+                },
+            },
+        },
+    })}
+    render={data => {
+        return <TabComponent dataField="brand.keyword" {...data} />;
+    }}
+/>
+```
+
+You can also take advantage of various ReactiveSearch components static method for generating query. So we our basically selecting a single tab so we can use `SingleList` static method to generate query as follow:
+
+```js
+<ReactiveComponent
+    componentId="Brand"
+    showFilter
+    defaultQuery={() => ({
+        ...SingleList.generateQueryOptions({
+            dataField: "brand.keyword",
+            size: 5,
+            sortBy: "count"
         })
-    }
+    })}
+    render={data => {
+        return <TabComponent dataField="brand.keyword" {...data} />;
+    }}
 />
 ```
 
+Here `TabComponent` is the component responsible to render the Tabs UI and handle switching tab and displaying selected filters. We are using `render` prop to display the component which provides a object with different params that can used in rendering UI and dispatching new query.Fow switching tab we will use `onClick` event on each tab to fire new query. We will make use of following param provided by `render` method: 
```

Comment:
Some grammar changes would help. Like "an object" instead of "a object", and re-read the sentences. I think we can phrase them better.

Reply:
`Fow switching` @jyash97 Can you run this through a grammar / typo checker once?